lib/ansible/module_utils/ovirt.py

```diff
@@ -0,0 +1,536 @@
+        entity=None,
+        action_condition=lambda e: e,
+        wait_condition=lambda e: e,
+        pre_action=lambda e: e,
+        post_action=lambda e: None,
+        search_params=None,
+        **kwargs
+    ):
+        """
+        This method is executed when we want to change the state of some oVirt
+        entity. The action to be executed on oVirt service is specified by
+        `action` parameter. Whether the action should be executed can be
+        specified by passing `action_condition` parameter. Which state the entity
+        should have after executing the action can be specified by passing
+        `wait_condition` parameter.
+
+        Additional task can be executed before action on entity is executed
+        by passing `pre_action` parameter to method. Another task can be
+        executed after action is executed, by passing `post_action` parameter
+        to method.
+
+        :param action: Action which should be executed on service.
+        :param entity: Entity we want to run action on.
+        :param action_condition: Function which is executed when checking if action should be executed.
+        :param wait_condition: Function which is executed when waiting on result state.
+        :param pre_action: Function which is executed before running the action.
+        :param post_action: Function which is executed after running the action.
+        :param search_params: Dictionary of parameters to be used for search.
+        :param kwargs: Additional parameters passed to action.
+        :return: Dictionary with values returned by Ansible module.
+        """
+        if entity is None:
+            entity = self.search_entity(search_params)
+
+        entity = pre_action(entity)
+
+        if entity is None:
+            self._module.fail_json(
+                msg="Entity not found, can't run action '{}'.".format(
+                    action
+                )
+            )
+
+        entity_service = self._service.service(entity.id)
+        entity = entity_service.get()
+        if action_condition(entity):
+            if not self._module.check_mode:
+                getattr(entity_service, action)(**kwargs)
+            self.changed = True
+
+        post_action(entity)
+
+        wait(
+            service=entity_service,
+            condition=wait_condition,
+            wait=self._module.params['wait'],
+            timeout=self._module.params['timeout'],
+        )
+        return {
+            'changed': self.changed,
+            'id': entity.id,
+            type(entity).__name__.lower(): get_dict_of_struct(entity),
+        }
+
+    def search_entity(self, search_params=None):
+        """
+        Always first try to search by `ID`, if ID isn't specified,
+        check if user constructed special search in `search_params`,
+        if not search by `name`.
+        """
+        entity = None
+
+        if 'id' in self._module.params and self._module.params['id'] is not None:
```

Comment:
I would actually assume most of the users will not be using ID but name. Don't think it matters that much though.